--- a/client/screens/map/RenderMapScreen.java
+++ b/client/screens/map/RenderMapScreen.java
@@ -1,9 +1,9 @@
-package com.unascribed.blockrenderer.fabric.client.screens.map;
+package com.unascribed.blockrenderer.forge.client.screens.map;
 
 import com.mojang.blaze3d.matrix.MatrixStack;
-import com.unascribed.blockrenderer.fabric.client.render.RenderManager;
-import com.unascribed.blockrenderer.fabric.client.render.Requests;
-import com.unascribed.blockrenderer.fabric.client.screens.item.EnterSizeScreen;
+import com.unascribed.blockrenderer.forge.client.render.RenderManager;
+import com.unascribed.blockrenderer.forge.client.render.Requests;
+import com.unascribed.blockrenderer.forge.client.screens.item.EnterSizeScreen;
 import com.unascribed.blockrenderer.render.map.MapDecorations;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.screen.Screen;
@@ -14,7 +14,7 @@
 import net.minecraft.world.storage.MapData;
 import org.jetbrains.annotations.Nullable;
 
-import static com.unascribed.blockrenderer.fabric.client.varia.Strings.translate;
+import static com.unascribed.blockrenderer.forge.client.varia.Strings.translate;
 
 @SuppressWarnings("NotNullFieldNotInitialized")
 public class RenderMapScreen extends EnterSizeScreen {
