--- a/BlockRenderer.java
+++ b/BlockRenderer.java
@@ -1,41 +1,26 @@
-package com.unascribed.blockrenderer.forge;
+package com.unascribed.blockrenderer.fabric;
 
 import com.unascribed.blockrenderer.InternalAPI;
 import com.unascribed.blockrenderer.Reference;
-import com.unascribed.blockrenderer.forge.client.init.Keybindings;
-import com.unascribed.blockrenderer.forge.client.proxy.ClientProxy;
-import com.unascribed.blockrenderer.forge.client.render.RenderManager;
-import com.unascribed.blockrenderer.forge.client.varia.Registries;
-import com.unascribed.blockrenderer.forge.client.varia.rendering.GL;
+import com.unascribed.blockrenderer.fabric.client.init.Keybindings;
+import com.unascribed.blockrenderer.fabric.client.render.RenderManager;
+import com.unascribed.blockrenderer.fabric.client.varia.rendering.GL;
 import com.unascribed.blockrenderer.varia.logging.Log;
 import com.unascribed.blockrenderer.varia.logging.Markers;
-import net.minecraftforge.common.MinecraftForge;
-import net.minecraftforge.event.TickEvent;
-import net.minecraftforge.fml.event.lifecycle.FMLClientSetupEvent;
-import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;
+import net.fabricmc.api.ClientModInitializer;
 
-public class BlockRenderer {
+public class BlockRenderer implements ClientModInitializer {
 
     public BlockRenderer() {
-        FMLJavaModLoadingContext.get().getModEventBus().addListener(BlockRenderer::onInitializeClient);
-        MinecraftForge.EVENT_BUS.addListener(BlockRenderer::onFrameStart);
-
-        Registries.clazzLoad();
-
         Log.info(Markers.ROOT, "Running Version: " + Reference.VERSION);
 
         InternalAPI.setGL(GL.INSTANCE);
         InternalAPI.setRenderManager(RenderManager.INSTANCE);
     }
 
-    private static void onInitializeClient(FMLClientSetupEvent event) {
+    @Override
+    public void onInitializeClient() {
         Keybindings.register();
     }
 
-    private static void onFrameStart(TickEvent.RenderTickEvent e) {
-        if (e.phase != TickEvent.Phase.START) return;
-
-        ClientProxy.onFrameStart();
     }
-
-}
