--- a/client/init/Keybindings.java
+++ b/client/init/Keybindings.java
@@ -1,15 +1,15 @@
-package com.unascribed.blockrenderer.forge.client.init;
+package com.unascribed.blockrenderer.fabric.client.init;
 
-import net.minecraft.client.settings.KeyBinding;
-import net.minecraftforge.fml.client.registry.ClientRegistry;
+import net.fabricmc.fabric.api.client.keybinding.v1.KeyBindingHelper;
+import net.minecraft.client.KeyMapping;
 import org.lwjgl.glfw.GLFW;
 
 public interface Keybindings {
 
-    KeyBinding render = new KeyBinding("key.block_renderer.render", GLFW.GLFW_KEY_GRAVE_ACCENT, "key.categories.block_renderer");
+    KeyMapping render = new KeyMapping("key.block_renderer.render", GLFW.GLFW_KEY_GRAVE_ACCENT, "key.categories.block_renderer");
 
     static void register() {
-        ClientRegistry.registerKeyBinding(render);
+        KeyBindingHelper.registerKeyBinding(render);
     }
 
 }
