--- a/client/screens/widgets/ItemButtonWidget.java
+++ b/client/screens/widgets/ItemButtonWidget.java
@@ -1,9 +1,9 @@
-package com.unascribed.blockrenderer.fabric.client.screens.widgets;
+package com.unascribed.blockrenderer.forge.client.screens.widgets;
 
 import com.mojang.blaze3d.matrix.MatrixStack;
 import com.unascribed.blockrenderer.InternalAPI;
-import com.unascribed.blockrenderer.fabric.client.varia.Registries;
-import com.unascribed.blockrenderer.fabric.client.varia.rendering.Display;
+import com.unascribed.blockrenderer.forge.client.varia.Registries;
+import com.unascribed.blockrenderer.forge.client.varia.rendering.Display;
 import com.unascribed.blockrenderer.varia.rendering.GLI;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.screen.Screen;
@@ -20,8 +20,8 @@
 public class ItemButtonWidget extends Button {
 
     public static final ResourceLocation CUSTOM_WIDGETS = new ResourceLocation("block_renderer:textures/gui/widgets.png");
-
     private final GLI GL = InternalAPI.getGL();
+
     private final Screen owner;
     private final Supplier<List<ITextComponent>> tooltip;
     private final ItemRenderer renderer;
@@ -38,7 +38,7 @@
         this.owner = owner;
         this.renderer = renderer;
         this.stack = stack;
-        this.isMap = stack.get().getItem() == Registries.MAP.getValue();
+        this.isMap = stack.get().getItem() == Registries.MAP.get();
     }
 
     @Override
