--- a/client/screens/item/EnterNamespaceScreen.java
+++ b/client/screens/item/EnterNamespaceScreen.java
@@ -1,12 +1,12 @@
-package com.unascribed.blockrenderer.forge.client.screens.item;
+package com.unascribed.blockrenderer.fabric.client.screens.item;
 
 import com.mojang.blaze3d.matrix.MatrixStack;
-import com.unascribed.blockrenderer.forge.client.render.Requests;
-import com.unascribed.blockrenderer.forge.client.render.manager.RenderManager;
-import com.unascribed.blockrenderer.forge.client.screens.widgets.HoverableTinyButtonWidget;
-import com.unascribed.blockrenderer.forge.client.screens.widgets.ItemButtonMultiWidget;
-import com.unascribed.blockrenderer.forge.client.varia.Registries;
-import com.unascribed.blockrenderer.forge.client.varia.Strings;
+import com.unascribed.blockrenderer.fabric.client.render.Requests;
+import com.unascribed.blockrenderer.fabric.client.render.manager.RenderManager;
+import com.unascribed.blockrenderer.fabric.client.screens.widgets.HoverableTinyButtonWidget;
+import com.unascribed.blockrenderer.fabric.client.screens.widgets.ItemButtonMultiWidget;
+import com.unascribed.blockrenderer.fabric.client.varia.Registries;
+import com.unascribed.blockrenderer.fabric.client.varia.Strings;
 import net.minecraft.client.gui.screen.Screen;
 import net.minecraft.client.gui.widget.TextFieldWidget;
 import net.minecraft.client.gui.widget.button.Button;
@@ -18,7 +18,7 @@
 import java.util.Collections;
 import java.util.function.Supplier;
 
-import static com.unascribed.blockrenderer.forge.client.varia.Strings.translate;
+import static com.unascribed.blockrenderer.fabric.client.varia.Strings.translate;
 
 /*
  * Note: Screen's get initialized in init
