--- a/client/screens/item/EnterSizeScreen.java
+++ b/client/screens/item/EnterSizeScreen.java
@@ -1,18 +1,18 @@
-package com.unascribed.blockrenderer.forge.client.screens.item;
+package com.unascribed.blockrenderer.fabric.client.screens.item;
 
-import com.unascribed.blockrenderer.forge.client.render.RenderManager;
-import com.unascribed.blockrenderer.forge.client.render.Requests;
-import com.unascribed.blockrenderer.forge.client.screens.widgets.HoverableTinyButtonWidget;
-import net.minecraft.client.gui.screen.Screen;
-import net.minecraft.client.gui.widget.button.Button;
-import net.minecraft.item.ItemStack;
-import net.minecraft.util.text.ITextComponent;
-import net.minecraft.util.text.TranslationTextComponent;
+import com.unascribed.blockrenderer.fabric.client.render.RenderManager;
+import com.unascribed.blockrenderer.fabric.client.render.Requests;
+import com.unascribed.blockrenderer.fabric.client.screens.widgets.HoverableTinyButtonWidget;
+import net.minecraft.client.gui.components.Button;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TranslatableComponent;
+import net.minecraft.world.item.ItemStack;
 import org.jetbrains.annotations.Nullable;
 
 public class EnterSizeScreen extends BaseItemScreen {
 
-    private static final TranslationTextComponent TITLE = new TranslationTextComponent("block_renderer.gui.renderItem");
+    private static final TranslatableComponent TITLE = new TranslatableComponent("block_renderer.gui.renderItem");
 
     protected final ItemStack stack;
     private final boolean enableSwitch;
@@ -21,7 +21,7 @@
         this(TITLE, old, stack, true);
     }
 
-    public EnterSizeScreen(ITextComponent title, @Nullable Screen old, ItemStack stack, boolean enableSwitch) {
+    public EnterSizeScreen(Component title, @Nullable Screen old, ItemStack stack, boolean enableSwitch) {
         super(title, old);
         this.stack = stack;
         this.enableSwitch = enableSwitch;
@@ -41,8 +41,8 @@
                         this,
                         width - 32,
                         height - 32,
-                        new TranslationTextComponent("block_renderer.gui.switch.bulk"),
-                        new TranslationTextComponent("block_renderer.gui.switch.bulk.tooltip"),
+                        new TranslatableComponent("block_renderer.gui.switch.bulk"),
+                        new TranslatableComponent("block_renderer.gui.switch.bulk.tooltip"),
                         button -> minecraft.setScreen(new EnterNamespaceScreen(old, stack))),
                 enableSwitch
         );
