--- a/client/screens/item/RenderAnimatedScreen.java
+++ b/client/screens/item/RenderAnimatedScreen.java
@@ -1,9 +1,9 @@
-package com.unascribed.blockrenderer.forge.client.screens.item;
+package com.unascribed.blockrenderer.fabric.client.screens.item;
 
-import com.unascribed.blockrenderer.forge.client.render.Requests;
-import com.unascribed.blockrenderer.forge.client.render.manager.RenderManager;
-import com.unascribed.blockrenderer.forge.client.screens.widgets.HoverableCheckboxWidget;
-import com.unascribed.blockrenderer.forge.client.screens.widgets.HoverableTextFieldWidget;
+import com.unascribed.blockrenderer.fabric.client.render.Requests;
+import com.unascribed.blockrenderer.fabric.client.render.manager.RenderManager;
+import com.unascribed.blockrenderer.fabric.client.screens.widgets.HoverableCheckboxWidget;
+import com.unascribed.blockrenderer.fabric.client.screens.widgets.HoverableTextFieldWidget;
 import net.minecraft.client.gui.screen.Screen;
 import net.minecraft.client.gui.widget.TextFieldWidget;
 import net.minecraft.client.gui.widget.button.Button;
@@ -12,7 +12,7 @@
 import net.minecraft.util.text.TranslationTextComponent;
 import org.jetbrains.annotations.Nullable;
 
-import static com.unascribed.blockrenderer.forge.client.varia.Strings.translate;
+import static com.unascribed.blockrenderer.fabric.client.varia.Strings.translate;
 
 @SuppressWarnings("NotNullFieldNotInitialized")
 public class RenderAnimatedScreen extends EnterSizeScreen {
