--- a/client/varia/rendering/GL.java
+++ b/client/varia/rendering/GL.java
@@ -1,10 +1,10 @@
-package com.unascribed.blockrenderer.fabric.client.varia.rendering;
+package com.unascribed.blockrenderer.forge.client.varia.rendering;
 
-import com.mojang.blaze3d.platform.Lighting;
-import com.mojang.blaze3d.platform.Window;
 import com.mojang.blaze3d.systems.RenderSystem;
 import com.unascribed.blockrenderer.varia.rendering.GLI;
+import net.minecraft.client.MainWindow;
 import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.RenderHelper;
 import org.intellij.lang.annotations.MagicConstant;
 import org.lwjgl.opengl.GL11;
 
@@ -16,7 +16,7 @@
     public static final GLI INSTANCE = new GL();
 
     private static final Minecraft client = Minecraft.getInstance();
-    private static final Window window = client.getWindow();
+    private static final MainWindow window = client.getWindow();
 
     /* ================================================================================================== Matrix ==== */
 
@@ -51,12 +51,12 @@
 
     @Override
     public void setupItemStackLighting() {
-        Lighting.setupFor3DItems();
+        RenderHelper.setupFor3DItems();
     }
 
     @Override
     public void displayLighting() {
-        Lighting.turnOff();
+        RenderHelper.turnOff();
     }
 
     /* =================================================================================================== State ==== */
@@ -139,7 +139,7 @@
     @Override
     public void setupOverlayRendering() {
         Minecraft client = Minecraft.getInstance();
-        Window window = client.getWindow();
+        MainWindow window = client.getWindow();
         double scaleFactor = window.getGuiScale();
 
         RenderSystem.clear(GL_DEPTH_BUFFER_BIT, Minecraft.ON_OSX);
