--- a/client/render/RenderManager.java
+++ b/client/render/RenderManager.java
@@ -1,6 +1,6 @@
-package com.unascribed.blockrenderer.forge.client.render;
+package com.unascribed.blockrenderer.fabric.client.render;
 
-import com.unascribed.blockrenderer.forge.client.render.report.Reporter;
+import com.unascribed.blockrenderer.fabric.client.render.report.Reporter;
 import com.unascribed.blockrenderer.render.BaseRenderManager;
 import com.unascribed.blockrenderer.render.IRequest;
 import net.minecraft.util.text.ITextComponent;
