--- a/client/proxy/ClientProxy.java
+++ b/client/proxy/ClientProxy.java
@@ -1,13 +1,11 @@
-package com.unascribed.blockrenderer.fabric.client.proxy;
+package com.unascribed.blockrenderer.forge.client.proxy;
 
-import com.unascribed.blockrenderer.fabric.client.init.Keybindings;
-import com.unascribed.blockrenderer.fabric.client.render.Requests;
-import com.unascribed.blockrenderer.fabric.client.render.manager.RenderManager;
-import com.unascribed.blockrenderer.fabric.client.screens.SelectionScreen;
-import com.unascribed.blockrenderer.fabric.client.screens.item.EnterSizeScreen;
-import com.unascribed.blockrenderer.fabric.client.varia.Strings;
-import com.unascribed.blockrenderer.fabric.mixin.accessor.IHoveredSlot;
-import net.fabricmc.fabric.api.client.keybinding.v1.KeyBindingHelper;
+import com.unascribed.blockrenderer.forge.client.init.Keybindings;
+import com.unascribed.blockrenderer.forge.client.render.Requests;
+import com.unascribed.blockrenderer.forge.client.render.manager.RenderManager;
+import com.unascribed.blockrenderer.forge.client.screens.SelectionScreen;
+import com.unascribed.blockrenderer.forge.client.screens.item.EnterSizeScreen;
+import com.unascribed.blockrenderer.forge.client.varia.Strings;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.recipebook.IRecipeShownListener;
 import net.minecraft.client.gui.recipebook.RecipeBookGui;
@@ -20,7 +18,7 @@
 import net.minecraft.item.ItemStack;
 import org.lwjgl.glfw.GLFW;
 
-import static com.unascribed.blockrenderer.fabric.client.varia.Strings.translate;
+import static com.unascribed.blockrenderer.forge.client.varia.Strings.translate;
 
 public class ClientProxy {
 
@@ -40,10 +38,10 @@
         Minecraft client = Minecraft.getInstance();
         Slot hovered = null;
         Screen currentScreen = client.currentScreen;
-        boolean isContainerScreen = currentScreen instanceof IHoveredSlot;
+        boolean isContainerScreen = currentScreen instanceof ContainerScreen<?>;
 
         // Intellij Bork-ing wants explicit null check.
-        if (currentScreen != null && isContainerScreen) hovered = ((IHoveredSlot) currentScreen).block_renderer$accessor$hoveredSlot();
+        if (currentScreen != null && isContainerScreen) hovered = ((ContainerScreen<?>) currentScreen).getSlotUnderMouse();
 
         if (Screen.hasControlDown()) {
             PlayerEntity player = client.player;
@@ -119,7 +117,7 @@
         }
 
         /* Actually Check to see if the key is down */
-        InputMappings.Input key = KeyBindingHelper.getBoundKeyOf(Keybindings.render);
+        InputMappings.Input key = Keybindings.render.getKey();
 
         if (key.getType() == InputMappings.Type.MOUSE) {
             return GLFW.glfwGetMouseButton(client.getMainWindow().getHandle(), key.getKeyCode()) == GLFW.GLFW_PRESS;
