--- a/client/screens/BaseScreen.java
+++ b/client/screens/BaseScreen.java
@@ -1,8 +1,8 @@
-package com.unascribed.blockrenderer.fabric.client.screens;
+package com.unascribed.blockrenderer.forge.client.screens;
 
 import com.mojang.blaze3d.matrix.MatrixStack;
-import com.unascribed.blockrenderer.fabric.client.screens.widgets.UpdateableSliderWidget;
-import com.unascribed.blockrenderer.fabric.client.screens.widgets.options.ExtendedSliderPercentageOption;
+import com.unascribed.blockrenderer.forge.client.screens.widgets.UpdateableSliderWidget;
+import com.unascribed.blockrenderer.forge.client.screens.widgets.options.ExtendedSliderPercentageOption;
 import com.unascribed.blockrenderer.varia.Maths;
 import net.minecraft.client.GameSettings;
 import net.minecraft.client.gui.screen.Screen;
@@ -12,7 +12,7 @@
 import org.jetbrains.annotations.Nullable;
 import org.lwjgl.glfw.GLFW;
 
-import static com.unascribed.blockrenderer.fabric.client.varia.StringUtils.translate;
+import static com.unascribed.blockrenderer.forge.client.varia.StringUtils.translate;
 
 @SuppressWarnings("NotNullFieldNotInitialized")
 public abstract class BaseScreen extends Screen {
