--- a/client/screens/SelectionScreen.java
+++ b/client/screens/SelectionScreen.java
@@ -1,12 +1,12 @@
-package com.unascribed.blockrenderer.fabric.client.screens;
+package com.unascribed.blockrenderer.forge.client.screens;
 
 import com.mojang.blaze3d.matrix.MatrixStack;
-import com.unascribed.blockrenderer.fabric.client.screens.item.EnterNamespaceScreen;
-import com.unascribed.blockrenderer.fabric.client.screens.item.RenderAnimatedScreen;
-import com.unascribed.blockrenderer.fabric.client.screens.map.RenderMapScreen;
-import com.unascribed.blockrenderer.fabric.client.screens.widgets.ItemButtonWidget;
-import com.unascribed.blockrenderer.fabric.client.varia.Identifiers;
-import com.unascribed.blockrenderer.fabric.client.varia.Registries;
+import com.unascribed.blockrenderer.forge.client.screens.item.EnterNamespaceScreen;
+import com.unascribed.blockrenderer.forge.client.screens.item.RenderAnimatedScreen;
+import com.unascribed.blockrenderer.forge.client.screens.map.RenderMapScreen;
+import com.unascribed.blockrenderer.forge.client.screens.widgets.ItemButtonWidget;
+import com.unascribed.blockrenderer.forge.client.varia.Identifiers;
+import com.unascribed.blockrenderer.forge.client.varia.Registries;
 import net.minecraft.client.gui.screen.Screen;
 import net.minecraft.item.Item;
 import net.minecraft.item.ItemStack;
