--- a/client/render/item/DefaultPngItemStackHandler.java
+++ b/client/render/item/DefaultPngItemStackHandler.java
@@ -1,8 +1,8 @@
-package com.unascribed.blockrenderer.forge.client.render.item;
+package com.unascribed.blockrenderer.fabric.client.render.item;
 
 import com.unascribed.blockrenderer.render.request.lambda.ImageHandler;
 import com.unascribed.blockrenderer.varia.Files;
-import net.minecraft.item.ItemStack;
+import net.minecraft.world.item.ItemStack;
 
 import java.awt.image.BufferedImage;
 import java.io.File;
