--- a/client/render/item/DefaultPngItemStackHandler.java
+++ b/client/render/item/DefaultPngItemStackHandler.java
@@ -1,6 +1,6 @@
-package com.unascribed.blockrenderer.forge.client.render.item;
+package com.unascribed.blockrenderer.fabric.client.render.item;
 
-import com.unascribed.blockrenderer.forge.client.varia.StringUtils;
+import com.unascribed.blockrenderer.fabric.client.varia.StringUtils;
 import com.unascribed.blockrenderer.render.request.lambda.ImageHandler;
 import com.unascribed.blockrenderer.varia.Files;
 import net.minecraft.item.BlockItem;
@@ -40,7 +40,7 @@
     @Override
     public void run() {
         Style open = Style.EMPTY
-                .applyFormatting(TextFormatting.GOLD)
+                .setFormatting(TextFormatting.GOLD)
                 .setClickEvent(new ClickEvent(ClickEvent.Action.OPEN_FILE, folder.getAbsolutePath()));
 
         StringUtils.addMessage(new StringTextComponent("> Finished Rendering").setStyle(open));
@@ -54,7 +54,7 @@
             case 1:
                 ItemGroup group = value.getItem().getGroup();
                 if (group == null) return result;
-                return StringUtils.sanitize(group.getPath()) + "/" + result;
+                return StringUtils.sanitize(group.getGroupName()) + "/" + result;
             case 2:
                 return (value.getItem() instanceof BlockItem ? "blocks" : "items") + "/" + result;
         }
