--- a/client/render/item/ItemStackRenderer.java
+++ b/client/render/item/ItemStackRenderer.java
@@ -1,18 +1,19 @@
-package com.unascribed.blockrenderer.forge.client.render.item;
+package com.unascribed.blockrenderer.fabric.client.render.item;
 
-import com.unascribed.blockrenderer.forge.client.varia.Identifiers;
-import com.unascribed.blockrenderer.forge.client.varia.rendering.GL;
+
+import com.mojang.blaze3d.platform.Window;
+import com.unascribed.blockrenderer.fabric.client.varia.Identifiers;
+import com.unascribed.blockrenderer.fabric.client.varia.rendering.GL;
 import com.unascribed.blockrenderer.render.IAnimatedRenderer;
 import com.unascribed.blockrenderer.render.request.lambda.ImageHandler;
 import com.unascribed.blockrenderer.varia.Images;
 import com.unascribed.blockrenderer.varia.Maths;
 import com.unascribed.blockrenderer.varia.debug.Debug;
 import com.unascribed.blockrenderer.varia.rendering.TileRenderer;
-import net.minecraft.client.MainWindow;
+import net.minecraft.Util;
 import net.minecraft.client.Minecraft;
-import net.minecraft.client.renderer.ItemRenderer;
-import net.minecraft.item.ItemStack;
-import net.minecraft.util.Util;
+import net.minecraft.client.renderer.entity.ItemRenderer;
+import net.minecraft.world.item.ItemStack;
 import org.jetbrains.annotations.Nullable;
 
 import java.util.function.LongSupplier;
@@ -36,7 +37,7 @@
     public void setup(ItemStackParameters parameters) {
         Debug.push("item/setup");
 
-        MainWindow window = Minecraft.getInstance().getWindow();
+        Window window = Minecraft.getInstance().getWindow();
         int displayWidth = window.getWidth();
         int displayHeight = window.getHeight();
 
@@ -80,7 +81,7 @@
         LongSupplier oldSupplier = Util.timeSource;
         Util.timeSource = () -> nano;
 
-        Minecraft.getInstance().textureManager.tick();
+        Minecraft.getInstance().getTextureManager().tick();
 
         do {
             tr.beginTile();
