--- a/client/render/item/ItemStackRenderer.java
+++ b/client/render/item/ItemStackRenderer.java
@@ -1,7 +1,7 @@
-package com.unascribed.blockrenderer.forge.client.render.item;
+package com.unascribed.blockrenderer.fabric.client.render.item;
 
-import com.unascribed.blockrenderer.forge.client.varia.Identifiers;
-import com.unascribed.blockrenderer.forge.client.varia.rendering.GL;
+import com.unascribed.blockrenderer.fabric.client.varia.Identifiers;
+import com.unascribed.blockrenderer.fabric.client.varia.rendering.GL;
 import com.unascribed.blockrenderer.render.IAnimatedRenderer;
 import com.unascribed.blockrenderer.render.item.ItemStackParameters;
 import com.unascribed.blockrenderer.render.request.lambda.ImageHandler;
