--- a/client/screens/map/RenderMapScreen.java
+++ b/client/screens/map/RenderMapScreen.java
@@ -1,28 +1,28 @@
-package com.unascribed.blockrenderer.forge.client.screens.map;
+package com.unascribed.blockrenderer.fabric.client.screens.map;
 
-import com.mojang.blaze3d.matrix.MatrixStack;
-import com.unascribed.blockrenderer.forge.client.render.RenderManager;
-import com.unascribed.blockrenderer.forge.client.render.Requests;
-import com.unascribed.blockrenderer.forge.client.screens.item.EnterSizeScreen;
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.unascribed.blockrenderer.fabric.client.render.RenderManager;
+import com.unascribed.blockrenderer.fabric.client.render.Requests;
+import com.unascribed.blockrenderer.fabric.client.screens.item.EnterSizeScreen;
 import com.unascribed.blockrenderer.render.map.MapDecorations;
 import net.minecraft.client.Minecraft;
-import net.minecraft.client.gui.screen.Screen;
-import net.minecraft.client.gui.widget.button.Button;
-import net.minecraft.item.FilledMapItem;
-import net.minecraft.item.ItemStack;
-import net.minecraft.util.text.TranslationTextComponent;
-import net.minecraft.world.storage.MapData;
+import net.minecraft.client.gui.components.Button;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.network.chat.TranslatableComponent;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.MapItem;
+import net.minecraft.world.level.saveddata.maps.MapItemSavedData;
 import org.jetbrains.annotations.Nullable;
 
-import static com.unascribed.blockrenderer.forge.client.varia.Strings.translate;
+import static com.unascribed.blockrenderer.fabric.client.varia.Strings.translate;
 
 @SuppressWarnings("NotNullFieldNotInitialized")
 public class RenderMapScreen extends EnterSizeScreen {
 
-    private static final TranslationTextComponent TITLE = translate("block_renderer.gui.map");
+    private static final TranslatableComponent TITLE = translate("block_renderer.gui.map");
 
     @Nullable
-    private final MapData data;
+    private final MapItemSavedData data;
     protected Button decorationsButton;
     protected MapDecorations decorations = MapDecorations.DEFAULT;
 
@@ -30,7 +30,7 @@
         super(TITLE, old, stack, false);
 
         minecraft = Minecraft.getInstance();
-        data = minecraft.level != null ? FilledMapItem.getSavedData(stack, minecraft.level) : null;
+        data = minecraft.level != null ? MapItem.getSavedData(stack, minecraft.level) : null;
     }
 
     @Override
@@ -58,7 +58,7 @@
     }
 
     @Override
-    public void render(MatrixStack stack, int mouseX, int mouseY, float partialTicks) {
+    public void render(PoseStack stack, int mouseX, int mouseY, float partialTicks) {
         assert minecraft != null;
 
         super.render(stack, mouseX, mouseY, partialTicks);
