--- a/client/render/request/BulkRenderingRequest.java
+++ b/client/render/request/BulkRenderingRequest.java
@@ -1,7 +1,7 @@
-package com.unascribed.blockrenderer.forge.client.render.request;
+package com.unascribed.blockrenderer.fabric.client.render.request;
 
-import com.unascribed.blockrenderer.forge.client.render.report.ReporterBulk;
-import com.unascribed.blockrenderer.forge.client.screens.progress.ProgressLoader;
+import com.unascribed.blockrenderer.fabric.client.render.report.ReporterBulk;
+import com.unascribed.blockrenderer.fabric.client.screens.progress.ProgressLoader;
 import com.unascribed.blockrenderer.render.ILoader;
 import com.unascribed.blockrenderer.render.IRenderer;
 import com.unascribed.blockrenderer.render.request.BaseBulkRenderingRequest;
