--- a/client/render/request/RenderingRequest.java
+++ b/client/render/request/RenderingRequest.java
@@ -1,6 +1,6 @@
-package com.unascribed.blockrenderer.forge.client.render.request;
+package com.unascribed.blockrenderer.fabric.client.render.request;
 
-import com.unascribed.blockrenderer.forge.client.render.RenderManager;
+import com.unascribed.blockrenderer.fabric.client.render.RenderManager;
 import com.unascribed.blockrenderer.render.IRenderer;
 import com.unascribed.blockrenderer.render.IRequest;
 import com.unascribed.blockrenderer.render.request.lambda.ImageHandler;
