--- a/client/varia/Identifiers.java
+++ b/client/varia/Identifiers.java
@@ -1,16 +1,13 @@
-package com.unascribed.blockrenderer.forge.client.varia;
+package com.unascribed.blockrenderer.fabric.client.varia;
 
-import net.minecraft.util.ResourceLocation;
-import net.minecraftforge.registries.IForgeRegistryEntry;
+import net.minecraft.core.Registry;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.item.Item;
 
 public interface Identifiers {
 
-    static ResourceLocation get(IForgeRegistryEntry<?> value) {
-        ResourceLocation identifier = value.getRegistryName();
-
-        if (identifier == null) return new ResourceLocation("air");
-
-        return identifier;
+    static ResourceLocation get(Item value) {
+        return Registry.ITEM.getKey(value);
     }
 
 }
