--- a/client/varia/Styles.java
+++ b/client/varia/Styles.java
@@ -1,9 +1,9 @@
-package com.unascribed.blockrenderer.forge.client.varia;
+package com.unascribed.blockrenderer.fabric.client.varia;
 
-import net.minecraft.util.text.TextFormatting;
+import net.minecraft.ChatFormatting;
 
 public interface Styles {
 
-    TextFormatting GOLD = TextFormatting.GOLD;
+    ChatFormatting GOLD = ChatFormatting.GOLD;
 
 }
