--- a/client/varia/Identifiers.java
+++ b/client/varia/Identifiers.java
@@ -1,13 +1,16 @@
-package com.unascribed.blockrenderer.fabric.client.varia;
+package com.unascribed.blockrenderer.forge.client.varia;
 
-import net.minecraft.item.Item;
 import net.minecraft.util.ResourceLocation;
-import net.minecraft.util.registry.Registry;
+import net.minecraftforge.registries.IForgeRegistryEntry;
 
 public interface Identifiers {
 
-    static ResourceLocation get(Item value) {
-        return Registry.ITEM.getKey(value);
+    static ResourceLocation get(IForgeRegistryEntry<?> value) {
+        ResourceLocation identifier = value.getRegistryName();
+
+        if (identifier == null) return new ResourceLocation("air");
+
+        return identifier;
     }
 
 }
