--- a/client/varia/MiscUtils.java
+++ b/client/varia/MiscUtils.java
@@ -1,4 +1,4 @@
-package com.unascribed.blockrenderer.fabric.client.varia;
+package com.unascribed.blockrenderer.forge.client.varia;
 
 import com.google.common.collect.Lists;
 import com.unascribed.blockrenderer.varia.logging.Log;
@@ -11,7 +11,7 @@
 import net.minecraft.item.Items;
 import net.minecraft.util.NonNullList;
 import net.minecraft.util.ResourceLocation;
-import net.minecraft.util.registry.Registry;
+import net.minecraftforge.registries.ForgeRegistries;
 import org.lwjgl.glfw.GLFW;
 
 import java.util.List;
@@ -27,12 +27,12 @@
         List<ItemStack> stacks = Lists.newArrayList();
         NonNullList<ItemStack> list = NonNullList.create();
 
-        for (ResourceLocation identifier : Registry.ITEM.keySet()) {
+        for (ResourceLocation identifier : ForgeRegistries.ITEMS.getKeys()) {
             if (identifier != null && namespaces.contains(identifier.getNamespace()) || namespaces.contains("*")) {
                 list.clear();
 
-                Item item = Registry.ITEM.getOrDefault(identifier);
-                if (item == Items.AIR) continue;
+                Item item = ForgeRegistries.ITEMS.getValue(identifier);
+                if (item == null || item == Items.AIR) continue;
 
                 try {
                     item.fillItemGroup(ItemGroup.SEARCH, list);
