--- a/client/screens/BaseScreen.java
+++ b/client/screens/BaseScreen.java
@@ -1,18 +1,18 @@
-package com.unascribed.blockrenderer.forge.client.screens;
+package com.unascribed.blockrenderer.fabric.client.screens;
 
-import com.mojang.blaze3d.matrix.MatrixStack;
-import com.unascribed.blockrenderer.forge.client.screens.widgets.UpdateableSliderWidget;
-import com.unascribed.blockrenderer.forge.client.screens.widgets.options.ExtendedSliderPercentageOption;
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.unascribed.blockrenderer.fabric.client.screens.widgets.UpdateableSliderWidget;
+import com.unascribed.blockrenderer.fabric.client.screens.widgets.options.ExtendedSliderPercentageOption;
 import com.unascribed.blockrenderer.varia.Maths;
-import net.minecraft.client.GameSettings;
-import net.minecraft.client.gui.screen.Screen;
-import net.minecraft.client.gui.widget.Widget;
-import net.minecraft.client.gui.widget.button.Button;
-import net.minecraft.util.text.ITextComponent;
+import net.minecraft.client.Options;
+import net.minecraft.client.gui.components.AbstractWidget;
+import net.minecraft.client.gui.components.Button;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.network.chat.Component;
 import org.jetbrains.annotations.Nullable;
 import org.lwjgl.glfw.GLFW;
 
-import static com.unascribed.blockrenderer.forge.client.varia.StringUtils.translate;
+import static com.unascribed.blockrenderer.fabric.client.varia.StringUtils.translate;
 
 @SuppressWarnings("NotNullFieldNotInitialized")
 public abstract class BaseScreen extends Screen {
@@ -28,7 +28,7 @@
     protected UpdateableSliderWidget slider;
     protected Button renderButton;
 
-    public BaseScreen(ITextComponent title, @Nullable Screen old) {
+    public BaseScreen(Component title, @Nullable Screen old) {
         super(title);
         this.old = old;
     }
@@ -79,7 +79,7 @@
     }
 
     @Override
-    public void render(MatrixStack stack, int mouseX, int mouseY, float partialTicks) {
+    public void render(PoseStack stack, int mouseX, int mouseY, float partialTicks) {
         assert minecraft != null;
 
         renderBackground(stack);
@@ -88,7 +88,7 @@
 
         drawCenteredString(stack, minecraft.font, title, width / 2, height / 6, -1);
 
-        for (Widget widget : buttons)
+        for (AbstractWidget widget : buttons)
             if (widget.isHovered())
                 widget.renderToolTip(stack, mouseX, mouseY);
 
@@ -99,12 +99,12 @@
 
     protected abstract void onRender(Button button);
 
-    public ITextComponent getSliderDisplay(GameSettings settings, ExtendedSliderPercentageOption option) {
+    public Component getSliderDisplay(Options settings, ExtendedSliderPercentageOption option) {
         int px = round(size);
         return option.getDisplayPrefix().copy().append(": " + px + "x" + px);
     }
 
-    protected <T extends Widget> T addButton(T button, boolean active) {
+    protected <T extends AbstractWidget> T addButton(T button, boolean active) {
         addButton(button);
 
         button.active = active;
