--- a/client/screens/SelectionScreen.java
+++ b/client/screens/SelectionScreen.java
@@ -1,26 +1,26 @@
-package com.unascribed.blockrenderer.forge.client.screens;
+package com.unascribed.blockrenderer.fabric.client.screens;
 
-import com.mojang.blaze3d.matrix.MatrixStack;
-import com.unascribed.blockrenderer.forge.client.screens.item.EnterNamespaceScreen;
-import com.unascribed.blockrenderer.forge.client.screens.item.RenderAnimatedScreen;
-import com.unascribed.blockrenderer.forge.client.screens.map.RenderMapScreen;
-import com.unascribed.blockrenderer.forge.client.screens.widgets.ItemButtonWidget;
-import com.unascribed.blockrenderer.forge.client.varia.Identifiers;
-import com.unascribed.blockrenderer.forge.client.varia.Registries;
-import net.minecraft.client.gui.screen.Screen;
-import net.minecraft.item.Item;
-import net.minecraft.item.ItemStack;
-import net.minecraft.util.ResourceLocation;
-import net.minecraft.util.text.TranslationTextComponent;
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.unascribed.blockrenderer.fabric.client.screens.item.EnterNamespaceScreen;
+import com.unascribed.blockrenderer.fabric.client.screens.item.RenderAnimatedScreen;
+import com.unascribed.blockrenderer.fabric.client.screens.map.RenderMapScreen;
+import com.unascribed.blockrenderer.fabric.client.screens.widgets.ItemButtonWidget;
+import com.unascribed.blockrenderer.fabric.client.varia.Identifiers;
+import com.unascribed.blockrenderer.fabric.client.varia.Registries;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.network.chat.TranslatableComponent;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
 import org.jetbrains.annotations.Nullable;
 
 import java.util.Objects;
 
-import static com.unascribed.blockrenderer.forge.client.varia.StringUtils.translate;
+import static com.unascribed.blockrenderer.fabric.client.varia.StringUtils.translate;
 
 public class SelectionScreen extends Screen {
 
-    private static final TranslationTextComponent TITLE = translate("block_renderer.gui.choose");
+    private static final TranslatableComponent TITLE = translate("block_renderer.gui.choose");
 
     @Nullable
     private final Screen old;
@@ -41,7 +41,7 @@
         addButton(new ItemButtonWidget(
                 this,
                 itemRenderer,
-                Registries.MAP.lazyMap(Item::getDefaultInstance),
+                Registries.mapLazy(Registries.MAP, Item::getDefaultInstance),
                 width / 2 - 64 - 12,
                 height / 2,
                 translate("block_renderer.gui.choose.map"),
@@ -55,7 +55,7 @@
         addButton(new ItemButtonWidget(
                 this,
                 itemRenderer,
-                Registries.DISPENSER.lazyMap(Item::getDefaultInstance),
+                Registries.mapLazy(Registries.DISPENSER, Item::getDefaultInstance),
                 width / 2,
                 height / 2,
                 translate("block_renderer.gui.choose.item"),
@@ -66,7 +66,7 @@
         addButton(new ItemButtonWidget(
                 this,
                 itemRenderer,
-                Registries.CUTTER.lazyMap(Item::getDefaultInstance),
+                Registries.mapLazy(Registries.CUTTER, Item::getDefaultInstance),
                 width / 2 + 64 + 12,
                 height / 2,
                 translate("block_renderer.gui.choose.animated"),
@@ -79,7 +79,7 @@
     }
 
     @Override
-    public void render(MatrixStack stack, int mouseX, int mouseY, float partialTicks) {
+    public void render(PoseStack stack, int mouseX, int mouseY, float partialTicks) {
         assert minecraft != null;
 
         renderBackground(stack);
