--- a/client/render/map/MapRenderer.java
+++ b/client/render/map/MapRenderer.java
@@ -1,7 +1,7 @@
-package com.unascribed.blockrenderer.fabric.client.render.map;
+package com.unascribed.blockrenderer.forge.client.render.map;
 
 import com.mojang.blaze3d.matrix.MatrixStack;
-import com.unascribed.blockrenderer.fabric.client.varia.rendering.GL;
+import com.unascribed.blockrenderer.forge.client.varia.rendering.GL;
 import com.unascribed.blockrenderer.render.IRenderer;
 import com.unascribed.blockrenderer.render.request.lambda.ImageHandler;
 import com.unascribed.blockrenderer.varia.Images;
@@ -70,10 +70,10 @@
         LongSupplier oldSupplier = Util.nanoTimeSupplier;
         Util.nanoTimeSupplier = () -> 0;
 
-        Minecraft.getInstance().getTextureManager().tick();
+        Minecraft.getInstance().textureManager.tick();
 
         IRenderTypeBuffer.Impl buffers = IRenderTypeBuffer.getImpl(Tessellator.getInstance().getBuffer());
-        try (MapItemRenderer renderer = new MapItemRenderer(Minecraft.getInstance().getTextureManager())) {
+        try (MapItemRenderer renderer = new MapItemRenderer(Minecraft.getInstance().textureManager)) {
             renderer.updateMapTexture(instance);
 
             do {
