--- a/client/render/map/MapRenderer.java
+++ b/client/render/map/MapRenderer.java
@@ -1,4 +1,4 @@
-package com.unascribed.blockrenderer.fabric.client.render.map;
+package com.unascribed.blockrenderer.forge.client.render.map;
 
 import com.mojang.blaze3d.matrix.MatrixStack;
 import com.unascribed.blockrenderer.InternalAPI;
@@ -25,6 +25,7 @@
 public class MapRenderer implements IRenderer<MapParameters, MapData> {
 
     private static final float MAP_SIZE = 128.0F;
+    
     private final GLI GL = InternalAPI.getGL();
 
     @Nullable
