--- a/client/screens/widgets/options/ExtendedSliderPercentageOption.java
+++ b/client/screens/widgets/options/ExtendedSliderPercentageOption.java
@@ -1,21 +1,20 @@
-package com.unascribed.blockrenderer.fabric.client.screens.widgets.options;
+package com.unascribed.blockrenderer.forge.client.screens.widgets.options;
 
-import net.minecraft.client.Options;
-import net.minecraft.client.ProgressOption;
-import net.minecraft.network.chat.Component;
+import net.minecraft.client.GameSettings;
+import net.minecraft.client.settings.SliderPercentageOption;
+import net.minecraft.util.text.ITextComponent;
 
 import java.util.function.BiConsumer;
 import java.util.function.BiFunction;
 import java.util.function.Function;
 
-public class ExtendedSliderPercentageOption extends ProgressOption {
+public class ExtendedSliderPercentageOption extends SliderPercentageOption {
 
-    public ExtendedSliderPercentageOption(String translationKey, double minValueIn, double maxValueIn, float stepSizeIn, Function<Options, Double> getter, BiConsumer<Options, Double> setter, BiFunction<Options, ExtendedSliderPercentageOption, Component> getDisplayString) {
+    public ExtendedSliderPercentageOption(String translationKey, double minValueIn, double maxValueIn, float stepSizeIn, Function<GameSettings, Double> getter, BiConsumer<GameSettings, Double> setter, BiFunction<GameSettings, ExtendedSliderPercentageOption, ITextComponent> getDisplayString) {
         super(translationKey, minValueIn, maxValueIn, stepSizeIn, getter, setter, (settings, option) -> getDisplayString.apply(settings, (ExtendedSliderPercentageOption) option));
     }
 
-    public Component getDisplayPrefix() {
+    public ITextComponent getDisplayPrefix() {
         return super.getCaption();
     }
-
 }
