--- a/client/render/item/DefaultGifItemStackHandler.java
+++ b/client/render/item/DefaultGifItemStackHandler.java
@@ -1,7 +1,7 @@
-package com.unascribed.blockrenderer.fabric.client.render.item;
+package com.unascribed.blockrenderer.forge.client.render.item;
 
 import com.unascribed.blockrenderer.varia.Files;
-import net.minecraft.world.item.ItemStack;
+import net.minecraft.item.ItemStack;
 import org.jetbrains.annotations.Nullable;
 
 import javax.imageio.stream.FileImageOutputStream;
