--- a/client/render/item/ItemRenderer.java
+++ b/client/render/item/ItemRenderer.java
@@ -1,10 +1,10 @@
-package com.unascribed.blockrenderer.fabric.client.render.item;
+package com.unascribed.blockrenderer.forge.client.render.item;
 
 import com.google.common.base.Joiner;
-import com.unascribed.blockrenderer.fabric.client.render.map.DefaultPngMapHandler;
-import com.unascribed.blockrenderer.fabric.client.render.map.MapRenderer;
-import com.unascribed.blockrenderer.fabric.client.varia.MiscUtils;
-import com.unascribed.blockrenderer.fabric.client.varia.StringUtils;
+import com.unascribed.blockrenderer.forge.client.render.map.DefaultPngMapHandler;
+import com.unascribed.blockrenderer.forge.client.render.map.MapRenderer;
+import com.unascribed.blockrenderer.forge.client.varia.MiscUtils;
+import com.unascribed.blockrenderer.forge.client.varia.StringUtils;
 import com.unascribed.blockrenderer.render.IRequest;
 import com.unascribed.blockrenderer.render.item.ItemStackParameters;
 import com.unascribed.blockrenderer.render.map.MapDecorations;
