--- a/client/render/item/ItemRenderer.java
+++ b/client/render/item/ItemRenderer.java
@@ -1,15 +1,15 @@
-package com.unascribed.blockrenderer.fabric.client.render.item;
+package com.unascribed.blockrenderer.forge.client.render.item;
 
 import com.google.common.base.Joiner;
-import com.unascribed.blockrenderer.fabric.client.render.map.DefaultPngMapHandler;
-import com.unascribed.blockrenderer.fabric.client.render.map.MapDecorations;
-import com.unascribed.blockrenderer.fabric.client.render.map.MapParameters;
-import com.unascribed.blockrenderer.fabric.client.render.map.MapRenderer;
-import com.unascribed.blockrenderer.fabric.client.render.request.AnimatedRenderingRequest;
-import com.unascribed.blockrenderer.fabric.client.render.request.BulkRenderingRequest;
-import com.unascribed.blockrenderer.fabric.client.render.request.RenderingRequest;
-import com.unascribed.blockrenderer.fabric.client.varia.MiscUtils;
-import com.unascribed.blockrenderer.fabric.client.varia.StringUtils;
+import com.unascribed.blockrenderer.forge.client.render.map.DefaultPngMapHandler;
+import com.unascribed.blockrenderer.forge.client.render.map.MapDecorations;
+import com.unascribed.blockrenderer.forge.client.render.map.MapParameters;
+import com.unascribed.blockrenderer.forge.client.render.map.MapRenderer;
+import com.unascribed.blockrenderer.forge.client.render.request.AnimatedRenderingRequest;
+import com.unascribed.blockrenderer.forge.client.render.request.BulkRenderingRequest;
+import com.unascribed.blockrenderer.forge.client.render.request.RenderingRequest;
+import com.unascribed.blockrenderer.forge.client.varia.MiscUtils;
+import com.unascribed.blockrenderer.forge.client.varia.StringUtils;
 import com.unascribed.blockrenderer.render.IRequest;
 import com.unascribed.blockrenderer.varia.Files;
 import net.minecraft.item.ItemStack;
