--- a/client/render/manager/RenderManager.java
+++ b/client/render/manager/RenderManager.java
@@ -1,6 +1,6 @@
-package com.unascribed.blockrenderer.forge.client.render.manager;
+package com.unascribed.blockrenderer.fabric.client.render.manager;
 
-import com.unascribed.blockrenderer.forge.client.render.report.Reporter;
+import com.unascribed.blockrenderer.fabric.client.render.report.Reporter;
 import com.unascribed.blockrenderer.render.IRequest;
 import com.unascribed.blockrenderer.render.manager.BaseRenderManager;
 import net.minecraft.util.text.ITextComponent;
@@ -10,7 +10,7 @@
 import java.util.PriorityQueue;
 import java.util.Queue;
 
-import static com.unascribed.blockrenderer.forge.client.varia.Strings.rawText;
+import static com.unascribed.blockrenderer.fabric.client.varia.Strings.rawText;
 
 public class RenderManager extends BaseRenderManager<ITextComponent> {
 
