--- a/client/render/map/MapRenderer.java
+++ b/client/render/map/MapRenderer.java
@@ -1,7 +1,7 @@
-package com.unascribed.blockrenderer.forge.client.render.map;
+package com.unascribed.blockrenderer.fabric.client.render.map;
 
 import com.mojang.blaze3d.matrix.MatrixStack;
-import com.unascribed.blockrenderer.forge.client.varia.rendering.GL;
+import com.unascribed.blockrenderer.fabric.client.varia.rendering.GL;
 import com.unascribed.blockrenderer.render.IRenderer;
 import com.unascribed.blockrenderer.render.map.MapDecorations;
 import com.unascribed.blockrenderer.render.map.MapParameters;
