--- a/client/varia/rendering/Display.java
+++ b/client/varia/rendering/Display.java
@@ -1,4 +1,4 @@
-package com.unascribed.blockrenderer.forge.client.varia.rendering;
+package com.unascribed.blockrenderer.fabric.client.varia.rendering;
 
 import com.mojang.blaze3d.matrix.MatrixStack;
 import com.unascribed.blockrenderer.InternalAPI;
@@ -12,7 +12,7 @@
 import net.minecraft.client.renderer.Tessellator;
 import net.minecraft.client.renderer.vertex.DefaultVertexFormats;
 import net.minecraft.util.text.ITextComponent;
-import net.minecraftforge.fml.client.gui.GuiUtils;
+import net.minecraft.util.text.StringTextComponent;
 
 import java.util.List;
 
@@ -23,7 +23,10 @@
     private static final GLI GL = InternalAPI.getGL();
     private static final Minecraft client = Minecraft.getInstance();
     private static final FontRenderer font = client.fontRenderer;
+    private static final Screen helper = new Screen(new StringTextComponent("DUMMY")) {
 
+    };
+
     @Override
     public void drawRect(int x1, int y1, int x2, int y2, int color) {
         drawRect(new MatrixStack(), x1, y1, x2, y2, color);
@@ -47,7 +50,8 @@
     }
 
     public static void renderTooltip(Screen owner, MatrixStack stack, List<ITextComponent> tooltip, int x, int y) {
-        GuiUtils.drawHoveringText(stack, tooltip, x, y, owner.width, owner.height, -1, font);
+        helper.init(client, owner.width, owner.height);
+        helper.func_243308_b(stack, tooltip, x, y);
     }
 
     @Override
