--- a/client/render/report/Reporter.java
+++ b/client/render/report/Reporter.java
@@ -1,12 +1,12 @@
-package com.unascribed.blockrenderer.forge.client.render.report;
+package com.unascribed.blockrenderer.fabric.client.render.report;
 
-import com.unascribed.blockrenderer.forge.client.varia.rendering.Display;
+import com.unascribed.blockrenderer.fabric.client.varia.rendering.Display;
 import com.unascribed.blockrenderer.render.report.BaseReporter;
-import net.minecraft.util.text.ITextComponent;
+import net.minecraft.network.chat.Component;
 
-import static com.unascribed.blockrenderer.forge.client.varia.Strings.rawText;
+import static com.unascribed.blockrenderer.fabric.client.varia.Strings.rawText;
 
-public class Reporter extends BaseReporter<ITextComponent> {
+public class Reporter extends BaseReporter<Component> {
 
     public static Reporter INSTANCE = new Reporter();
 
@@ -25,7 +25,7 @@
     }
 
     @Override
-    public ITextComponent getProgress() {
+    public Component getProgress() {
         return rawText(getProgressString());
     }
 
