--- a/client/render/report/Reporter.java
+++ b/client/render/report/Reporter.java
@@ -1,15 +1,15 @@
-package com.unascribed.blockrenderer.forge.client.render.report;
+package com.unascribed.blockrenderer.fabric.client.render.report;
 
-import com.unascribed.blockrenderer.forge.client.varia.rendering.Display;
-import com.unascribed.blockrenderer.forge.client.varia.rendering.GL;
+import com.unascribed.blockrenderer.fabric.client.varia.rendering.Display;
+import com.unascribed.blockrenderer.fabric.client.varia.rendering.GL;
 import com.unascribed.blockrenderer.render.report.BaseReporter;
 import com.unascribed.blockrenderer.varia.Maths;
 import com.unascribed.blockrenderer.varia.Time;
 import com.unascribed.blockrenderer.varia.debug.Debug;
-import net.minecraft.util.text.ITextComponent;
-import net.minecraft.util.text.StringTextComponent;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TextComponent;
 
-public class Reporter extends BaseReporter<ITextComponent> {
+public class Reporter extends BaseReporter<Component> {
 
     private static final int DARK_GREEN = 0xFF001100;
     private static final int LIGHT_GREEN = 0xFF55FF55;
@@ -19,7 +19,7 @@
     private static long lastRender = System.nanoTime();
 
     private Reporter() {
-        super(new StringTextComponent("Rendering"));
+        super(new TextComponent("Rendering"));
     }
 
     @Override
@@ -33,8 +33,8 @@
     }
 
     @Override
-    public ITextComponent getProgress() {
-        return new StringTextComponent(getProgressString());
+    public Component getProgress() {
+        return new TextComponent(getProgressString());
     }
 
     @Override
