--- a/client/render/report/Reporter.java
+++ b/client/render/report/Reporter.java
@@ -1,16 +1,16 @@
-package com.unascribed.blockrenderer.forge.client.render.report;
+package com.unascribed.blockrenderer.fabric.client.render.report;
 
-import com.unascribed.blockrenderer.forge.client.varia.rendering.Display;
+import com.unascribed.blockrenderer.fabric.client.varia.rendering.Display;
 import com.unascribed.blockrenderer.render.report.BaseReporter;
-import net.minecraft.util.text.ITextComponent;
-import net.minecraft.util.text.StringTextComponent;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TextComponent;
 
-public class Reporter extends BaseReporter<ITextComponent> {
+public class Reporter extends BaseReporter<Component> {
 
     public static Reporter INSTANCE = new Reporter();
 
     private Reporter() {
-        super(new StringTextComponent("Rendering"), Display.INSTANCE);
+        super(new TextComponent("Rendering"), Display.INSTANCE);
     }
 
     @Override
@@ -24,8 +24,8 @@
     }
 
     @Override
-    public ITextComponent getProgress() {
-        return new StringTextComponent(getProgressString());
+    public Component getProgress() {
+        return new TextComponent(getProgressString());
     }
 
 }
