--- a/client/screens/item/BaseItemScreen.java
+++ b/client/screens/item/BaseItemScreen.java
@@ -1,15 +1,15 @@
-package com.unascribed.blockrenderer.fabric.client.screens.item;
+package com.unascribed.blockrenderer.forge.client.screens.item;
 
-import com.unascribed.blockrenderer.fabric.client.screens.BaseScreen;
-import com.unascribed.blockrenderer.fabric.client.screens.widgets.HoverableCheckboxWidget;
-import com.unascribed.blockrenderer.fabric.client.screens.widgets.HoverableTinyButtonWidget;
+import com.unascribed.blockrenderer.forge.client.screens.BaseScreen;
+import com.unascribed.blockrenderer.forge.client.screens.widgets.HoverableCheckboxWidget;
+import com.unascribed.blockrenderer.forge.client.screens.widgets.HoverableTinyButtonWidget;
 import net.minecraft.client.gui.screen.Screen;
 import net.minecraft.client.gui.widget.button.Button;
 import net.minecraft.client.gui.widget.button.CheckboxButton;
 import net.minecraft.util.text.ITextComponent;
 import org.jetbrains.annotations.Nullable;
 
-import static com.unascribed.blockrenderer.fabric.client.varia.Strings.translate;
+import static com.unascribed.blockrenderer.forge.client.varia.Strings.translate;
 
 /*
  * Note: Screen's get initialized in init
