--- a/client/screens/item/EnterSizeScreen.java
+++ b/client/screens/item/EnterSizeScreen.java
@@ -1,8 +1,8 @@
-package com.unascribed.blockrenderer.fabric.client.screens.item;
+package com.unascribed.blockrenderer.forge.client.screens.item;
 
-import com.unascribed.blockrenderer.fabric.client.render.Requests;
-import com.unascribed.blockrenderer.fabric.client.render.manager.RenderManager;
-import com.unascribed.blockrenderer.fabric.client.screens.widgets.HoverableTinyButtonWidget;
+import com.unascribed.blockrenderer.forge.client.render.Requests;
+import com.unascribed.blockrenderer.forge.client.render.manager.RenderManager;
+import com.unascribed.blockrenderer.forge.client.screens.widgets.HoverableTinyButtonWidget;
 import net.minecraft.client.gui.screen.Screen;
 import net.minecraft.client.gui.widget.button.Button;
 import net.minecraft.item.ItemStack;
@@ -10,7 +10,7 @@
 import net.minecraft.util.text.TranslationTextComponent;
 import org.jetbrains.annotations.Nullable;
 
-import static com.unascribed.blockrenderer.fabric.client.varia.Strings.translate;
+import static com.unascribed.blockrenderer.forge.client.varia.Strings.translate;
 
 public class EnterSizeScreen extends BaseItemScreen {
 
