--- a/client/screens/item/EnterSizeScreen.java
+++ b/client/screens/item/EnterSizeScreen.java
@@ -1,20 +1,20 @@
-package com.unascribed.blockrenderer.fabric.client.screens.item;
+package com.unascribed.blockrenderer.forge.client.screens.item;
 
-import com.unascribed.blockrenderer.fabric.client.render.RenderManager;
-import com.unascribed.blockrenderer.fabric.client.render.Requests;
-import com.unascribed.blockrenderer.fabric.client.screens.widgets.HoverableTinyButtonWidget;
-import net.minecraft.client.gui.components.Button;
-import net.minecraft.client.gui.screens.Screen;
-import net.minecraft.network.chat.Component;
-import net.minecraft.network.chat.TranslatableComponent;
-import net.minecraft.world.item.ItemStack;
+import com.unascribed.blockrenderer.forge.client.render.RenderManager;
+import com.unascribed.blockrenderer.forge.client.render.Requests;
+import com.unascribed.blockrenderer.forge.client.screens.widgets.HoverableTinyButtonWidget;
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.client.gui.widget.button.Button;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.TranslationTextComponent;
 import org.jetbrains.annotations.Nullable;
 
-import static com.unascribed.blockrenderer.fabric.client.varia.StringUtils.translate;
+import static com.unascribed.blockrenderer.forge.client.varia.StringUtils.translate;
 
 public class EnterSizeScreen extends BaseItemScreen {
 
-    private static final TranslatableComponent TITLE = translate("block_renderer.gui.renderItem");
+    private static final TranslationTextComponent TITLE = translate("block_renderer.gui.renderItem");
 
     protected final ItemStack stack;
     private final boolean enableSwitch;
@@ -23,7 +23,7 @@
         this(TITLE, old, stack, true);
     }
 
-    public EnterSizeScreen(Component title, @Nullable Screen old, ItemStack stack, boolean enableSwitch) {
+    public EnterSizeScreen(ITextComponent title, @Nullable Screen old, ItemStack stack, boolean enableSwitch) {
         super(title, old);
         this.stack = stack;
         this.enableSwitch = enableSwitch;
