--- a/client/screens/item/RenderAnimatedScreen.java
+++ b/client/screens/item/RenderAnimatedScreen.java
@@ -1,27 +1,27 @@
-package com.unascribed.blockrenderer.fabric.client.screens.item;
+package com.unascribed.blockrenderer.forge.client.screens.item;
 
-import com.unascribed.blockrenderer.fabric.client.render.RenderManager;
-import com.unascribed.blockrenderer.fabric.client.render.Requests;
-import com.unascribed.blockrenderer.fabric.client.screens.widgets.HoverableCheckboxWidget;
-import com.unascribed.blockrenderer.fabric.client.screens.widgets.HoverableTextFieldWidget;
-import net.minecraft.client.gui.components.Button;
-import net.minecraft.client.gui.components.Checkbox;
-import net.minecraft.client.gui.components.EditBox;
-import net.minecraft.client.gui.screens.Screen;
-import net.minecraft.network.chat.TranslatableComponent;
-import net.minecraft.world.item.ItemStack;
+import com.unascribed.blockrenderer.forge.client.render.RenderManager;
+import com.unascribed.blockrenderer.forge.client.render.Requests;
+import com.unascribed.blockrenderer.forge.client.screens.widgets.HoverableCheckboxWidget;
+import com.unascribed.blockrenderer.forge.client.screens.widgets.HoverableTextFieldWidget;
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.client.gui.widget.TextFieldWidget;
+import net.minecraft.client.gui.widget.button.Button;
+import net.minecraft.client.gui.widget.button.CheckboxButton;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.text.TranslationTextComponent;
 import org.jetbrains.annotations.Nullable;
 
-import static com.unascribed.blockrenderer.fabric.client.varia.StringUtils.translate;
+import static com.unascribed.blockrenderer.forge.client.varia.StringUtils.translate;
 
 @SuppressWarnings("NotNullFieldNotInitialized")
 public class RenderAnimatedScreen extends EnterSizeScreen {
 
-    private static final TranslatableComponent TITLE = translate("block_renderer.gui.renderAnimated");
+    private static final TranslationTextComponent TITLE = translate("block_renderer.gui.renderAnimated");
 
-    private Checkbox autoLoop;
+    private CheckboxButton autoLoop;
 
-    private EditBox length;
+    private TextFieldWidget length;
 
     private boolean isInteger;
 
