--- a/client/render/report/Reporter.java
+++ b/client/render/report/Reporter.java
@@ -1,7 +1,7 @@
-package com.unascribed.blockrenderer.fabric.client.render.report;
+package com.unascribed.blockrenderer.forge.client.render.report;
 
-import com.unascribed.blockrenderer.fabric.client.varia.rendering.Display;
-import com.unascribed.blockrenderer.fabric.client.varia.rendering.GL;
+import com.unascribed.blockrenderer.forge.client.varia.rendering.Display;
+import com.unascribed.blockrenderer.forge.client.varia.rendering.GL;
 import com.unascribed.blockrenderer.render.report.BaseReporter;
 import com.unascribed.blockrenderer.varia.Maths;
 import com.unascribed.blockrenderer.varia.debug.Debug;
